### PR TITLE
new Rising world java egg

### PR DESCRIPTION
**Pull request by eagled2**
# Description

The current java legacy egg for rising world doesn't work anymore. Found here: https://github.com/pelican-eggs/games-steamcmd/tree/9fa9938fe9bae42e0132e01724ffc109edc6b614/rising_world/legacy

I'm making a new egg instead of modifying it for a couple of reasons.

- The legacy java beta version in steam is now called java
- The default (live) version also uses java so that seems like the better version to use.
- preserve the old version in case others prefer it over my changes

 Compared to the reference egg mine has these changes:

-  uses the default version instead of the non-existent legacy beta (could have also changed to the java beta for an older build)
- since i'm not using beta that var was removed
- warn in description about not setting memory limit
- rename example properties file to the correct name during install
- rename example permissions folder during install
- more vars needed during world creation are now set

## Checklist for all submissions

* [ x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [ x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [ x] Have you tested and reviewed your changes with confidence that everything works?
* [ x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [ x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ x] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [ x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [ x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [ x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [ x] The egg was exported from the panel

### CODE REVIEW

**Comment by gOOvER** [comment]
you know, that java is deprecated?

**Comment by eagled2** [comment]
> you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.

well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.

**Comment by gOOvER** [comment]
> > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> 
> well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.

but its not longer under active development. Unity version should be used

File: README.md [hunk 1]
```diff
@@ -80,6 +80,7 @@
 * [Quake Live](quake_live)
 * [Resonite](resonite)
 * [Rising World](rising_world)
+  * [Java](rising_world/java)
```
Comment (gOOvER) [inline README.md:83]: then delete legacy ;) The Legacy egg is java

File: rising_world/java/egg-rising-world-java.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PANEL",
+    "meta": {
+        "version": "PLCN_v1",
+        "update_url": null
+    },
+    "exported_at": "2024-12-26T20:49:48+00:00",
+    "name": "Rising World Java",
+    "author": "eagled2@zoho.com",
```
Comment (gOOvER) [inline rising_world/java/egg-rising-world-java.json:9]: please respect MIT Licence. Author cant be changed. You only imported the egg. My egg is licence here under MIT. With the next commit under AGPL

**Comment by eagled2** [comment]
> > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> 
> well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.

I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. 
I tried the java version since it's what worked last time I tried, which proved simpler.

**Review by gOOvER — CHANGES_REQUESTED** [review]

**Comment by gOOvER** [comment]
> > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > 
> > 
> > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> 
> I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.

thats not the problem of the egg. Java is outdated and devs said; use Unity.

Also, dont change the Author.

**Comment by eagled2** [comment]
> > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > 
> > 
> > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> 
> but its not longer under active development. Unity version should be used.
> 
> I see no need to update the java egg. Use the dev branch with Unity, what is the version, which is under active development. Read this in their forum ;)

Noted Just thought id save someone some time if they try java for similar reasons.

**Comment by gOOvER** [comment]
> > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > 
> > > 
> > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > 
> > 
> > but its not longer under active development. Unity version should be used.
> > I see no need to update the java egg. Use the dev branch with Unity, what is the version, which is under active development. Read this in their forum ;)
> 
> Noted Just thought id save someone some time if they try java for similar reasons.

i watch the development ;) I wrote both eggs and know, what going on

**Comment by eagled2** [comment]
> > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > 
> > > 
> > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > 
> > 
> > I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.
> 
> thats not the problem of the egg. Java is outdated and devs said; use Unity.
> 
> Also, dont change the Author.

Since this is getting denied i guess there is no point fixing this?

**Comment by gOOvER** [comment]
> > > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > > 
> > > > 
> > > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > > 
> > > 
> > > I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.
> > 
> > 
> > thats not the problem of the egg. Java is outdated and devs said; use Unity.
> > Also, dont change the Author.
> 
> Since this is getting denied i guess there is no point fixing this?

Not for the Java version, which is deprecated

**Comment by eagled2** [comment]
> > > > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > > > 
> > > > > 
> > > > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > > > 
> > > > 
> > > > I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.
> > > 
> > > 
> > > thats not the problem of the egg. Java is outdated and devs said; use Unity.
> > > Also, dont change the Author.
> > 
> > 
> > Since this is getting denied i guess there is no point fixing this?
> 
> Not for the Java version, which is deprecated

Noted in the future leave the author the same if branching an existing egg.

**Comment by gOOvER** [comment]
> > > > > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > > > > 
> > > > > > 
> > > > > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > > > > 
> > > > > 
> > > > > I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.
> > > > 
> > > > 
> > > > thats not the problem of the egg. Java is outdated and devs said; use Unity.
> > > > Also, dont change the Author.
> > > 
> > > 
> > > Since this is getting denied i guess there is no point fixing this?
> > 
> > 
> > Not for the Java version, which is deprecated
> 
> Noted in the future leave the author the same if branching an existing egg.

Thats normal. All egg have a licence. you cant overtake an egg and disrespect the licence.

**Comment by eagled2** [comment]
> > > > > > > > you know, that java is deprecated? you should only use the new unity server. So this PR is not really needed.
> > > > > > > 
> > > > > > > 
> > > > > > > well the live version still uses java which tells me its not. you only get the unity version it you specify unity as the beta branch.
> > > > > > 
> > > > > > 
> > > > > > I actually tried the unity version first but even installing on a physical machine it shows the server is running and connected with no errors. But the server doesn't respond to a connection attempt from the game. Using java from the same machine with all the same ports it worked fine though. I tried the java version since it's what worked last time I tried, which proved simpler.
> > > > > 
> > > > > 
> > > > > thats not the problem of the egg. Java is outdated and devs said; use Unity.
> > > > > Also, dont change the Author.
> > > > 
> > > > 
> > > > Since this is getting denied i guess there is no point fixing this?
> > > 
> > > 
> > > Not for the Java version, which is deprecated
> > 
> > 
> > Noted in the future leave the author the same if branching an existing egg.
> 
> Thats normal. All eggs have a licence. you cant overtake an egg and disrespect the licence.

Appreciate the guidance. It's new to me.

**Pull request closed without merging**